sentry_sdk/integrations/django/__init__.py

```diff
@@ -539,6 +648,25 @@ def raw_data(self):
 
     def form(self):
         # type: () -> QueryDict
+        """
+        Clean form data of any sensitive post parameters defined in the django app
+        see https://docs.djangoproject.com/en/3.2/_modules/django/views/decorators/debug/
+        """
+        sensitive_post_parameters = getattr(
+            self.request, "sensitive_post_parameters", []
+        )
+        if sensitive_post_parameters:
+            cleansed_request = self.request.POST.copy()
+            if sensitive_post_parameters == "__ALL__":
+                # Cleanse all parameters.
+                for k in cleansed_request:
+                    cleansed_request[k] = SENSITIVE_DATA_SUBSTITUTE
+            else:
+                # Cleanse only the specified parameters.
+                for param in sensitive_post_parameters:
+                    if param in cleansed_request:
+                        cleansed_request[param] = SENSITIVE_DATA_SUBSTITUTE
+            return cleansed_request
         return self.request.POST
```

Comment:
We should try to rewrite this function more concisely; below is one way we might do that:
```suggestion
        # type: () -> dict[str, Any]
        """
        Clean form data of any sensitive post parameters defined in the django app
        see https://docs.djangoproject.com/en/3.2/_modules/django/views/decorators/debug/
        """
        sensitive_post_parameters = getattr(
            self.request, "sensitive_post_parameters", []
        )

        if sensitive_post_parameters == "__ALL__":
            sensitive_post_parameters = self.request.POST.keys()

        return {
            k: SENSITIVE_DATA_SUBSTITUTE if k in sensitive_post_parameters else v
            for k, v in self.request.POST.items()
        }
```